lib/linked_list.rb

```diff
@@ -18,128 +18,269 @@ def initialize
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1), because the new node is always add at the beginning of the lilnked list
+    # Space Complexity: O(1), because the variables used here is a constant 
     def add_first(value)
-      raise NotImplementedError
+      return @head = Node.new(value) if @head.nil?
+      add_node = Node.new(value)
+      add_node.next = @head
+      @head = add_node
+      return @head

```

Comment:
Just a suggestion this is a bit simpler.

```suggestion
      return @head = Node.new(value, @head)
```